src/NLog/Internal/ReflectionHelpers.cs

```diff
@@ -272,6 +272,20 @@ public static Assembly GetAssembly(this Type type)
             return typeInfo.Assembly;
 #else
             return type.Assembly;
+#endif
+        }
+
+        public static bool IsValidPublicProperty(this PropertyInfo p)
+        {
+            return p.CanRead && p.GetIndexParameters().Length == 0 && p.GetGetMethod() != null;
+        }
+
+        public static object GetPropertyValue(this PropertyInfo p, object instance)
+        {
+#if NET45
```

Comment:
this is a bit unexpected for me:

![image](https://user-images.githubusercontent.com/5808377/65542377-5b40fd00-df0f-11e9-9470-5cb62f665608.png)

https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.getvalue?view=netframework-4.8#System_Reflection_PropertyInfo_GetValue_System_Object_

Maybe it should be `!SILVERLIGHT` etc?

Reply:
See https://docs.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.getvalue?view=netframework-4.0#System_Reflection_PropertyInfo_GetValue_System_Object_

Reply:
OK, missed that. Thanks!